mp4parse_capi/src/lib.rs

```diff
@@ -741,13 +741,12 @@ fn create_sample_table(track: &Track) -> Option<Vec<mp4parser_indice>> {
                 }
             true
         });
-    if !ret {
+    if sample_table.is_empty() {
```

Comment:
I just wanted to confirm if this this what you want? `ret` carried the return value of `all()` so it was false if any `end_offset` was zero, with `is_empty()` is false if any `end_offset` *wasn't* zero. Effectively converting the `.all()` into an `.any()`.

If this is want you want, I suggest using a second `.map()` and moving the `cur_position` update and `sample_table.push()` into the conditional so the logic is simpler.